### PR TITLE
CORE-574 add ErrorFielder and ErrorKeyer optional interfaces

### PR DESCRIPTION
**Pull request by ehudc**
I'll be testing this by plugging it to Accounts service first.

### CODE REVIEW

**Comment by ehudc** [comment]
This afternoon I was debugging an odd occurrence where a grpcc call was kicking off twice in the logger instead of just once, will look further next week.

**Comment by ehudc** [comment]
Noticed the grpcc calls get delayed readouts in the tailed logs when running docker up services, but still are logged just fine.

File: grpc/interceptor/logger.go [hunk 1]
```diff
@@ -19,22 +20,33 @@ func Logger(l *logrus.Logger) grpc.UnaryServerInterceptor {
 	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
 		ctx = addLoggerToContext(l, ctx)
 		entry := logger.FromContext(ctx)
-		name := info.FullMethod
+
+		// regex to change /service.Service/Endpt -> Endpt
+		r := regexp.MustCompile(`\/([A-Za-z])\w+\.([A-Za-z])\w+\/`)
```
Comment (cored) [inline grpc/interceptor/logger.go]: @ehudc What do you think if you extract this as a const `FormatServiceEndPoint` ?

Comment (bobbytables) [inline grpc/interceptor/logger.go]: Typically `MustCompile` is done at a package level variable.

Comment (ehudc) [inline grpc/interceptor/logger.go]: you mean declare this in interceptor.go? ok then

Comment (ehudc) [inline grpc/interceptor/logger.go]: actually, this is a logger related var, maybe the logger.go

File: logger/logger.go [hunk 2]
```diff
@@ -20,6 +21,10 @@ var (
 	// LogWriter is where the logs go. By default this empty buffer blackholes
 	// logs. It must be an io.Writer such as os.Stdout
 	Writer io.Writer = ioutil.Discard
+
+	// regex to format grpc call logs
+	// "/service.Service/Endpt" -> "Endpt"
+	FormatServiceEndpoint = regexp.MustCompile(`\/([A-Za-z])\w+\.([A-Za-z])\w+\/`)
```
Comment (cored) [inline logger/logger.go]: @ehudc is there any reason to make this a global var instead of a constant? Doesn't look like you change this in any way.

Comment (ehudc) [inline logger/logger.go]: no reason, you guys suggested it go to the package level, I moved it to the package level with the other vars... I can make it a constant

**Pull request merged**